Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -281,6 +282,22 @@ public void Serialize<T>(IIonWriter writer, T item)
                 return;
             }
 
+            Type genericDictionaryType = item.GetType().GetInterfaces().FirstOrDefault(t => t.GetGenericTypeDefinition().IsAssignableTo(typeof(IDictionary<,>)));

```

Comment:
Why is this not as simple as `if (thing is IDictionary)`?

Reply:
Two reasons

1. We only support IDictionary<string, any>, and checking for IDictionary and then the generic types can lead to certain custom classes being attempted to serailize when they aren't supported. Something could implement IDictionary as well as another class that has generics, which is not what we want.
2. In the happy path, we'd need to access this anyways, so this reduces the duplication